synapse/push/bulk_push_rule_evaluator.py

```diff
@@ -342,8 +348,21 @@ async def _action_for_event_by_user(
             for user_id, level in notification_levels.items():
                 notification_levels[user_id] = int(level)
 
+        # Pull out the mentions field if it exists and trim the values to things
+        # that might be valid.
+        mentions_raw = event.content.get(EventContentFields.MSC3952_MENTIONS)
+        if isinstance(mentions_raw, list):
+            # Take the first 10 items, then strip out any non-string ones and convert
+            # to a tuple.
```

Comment:
Does the MSC impose a limit on the number of mentions?

We're ignoring the 11th mention and onwards here, rather than rejecting the event outright. I suppose that's probably a lot more practical, because otherwise you'd have to update the auth rules and define a new room version.

Reply:
This was removed from the PR.

Reply:
So it was. But to check, there's no limit on the number of mentions now?

There's a mention under ["abuse potential"](https://github.com/matrix-org/matrix-spec-proposals/blob/clokep/intentional-mentions/proposals/3952-intentional-mentions.md#abuse-potential) that moderation tooling can limit the number of mentions. I didn't follow that---what's the thinking? Is the `m.mentions` field redactable or replaceable somehow?

Reply:
> There's a mention under ["abuse potential"](https://github.com/matrix-org/matrix-spec-proposals/blob/clokep/intentional-mentions/proposals/3952-intentional-mentions.md?rgh-link-date=2023-01-27T00%3A36%3A06Z#abuse-potential) that moderation tooling can limit the number of mentions. I didn't follow that---what's the thinking? Is the `m.mentions` field redactable or replaceable somehow?

You use something like mjolnir to ban/kick people who are abusive or you use a Synapse module to reject events with too many mentions.

Reply:
Thanks. I'm a little surprised---I'd bias towards having some kind of upper limit in the spec---but I'll defer to whatever the MSC process decides.